lib/pymedphys/_experimental/streamlit/apps/mosaiq_to_csv.py

```diff
@@ -0,0 +1,251 @@
+# Copyright (C) 2021 Cancer Care Associates
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import pathlib
+
+from pymedphys._imports import numpy as np
+from pymedphys._imports import pandas as pd
+from pymedphys._imports import streamlit as st
+
+import pymedphys
+from pymedphys._streamlit import categories
+from pymedphys._streamlit.utilities import config as st_config
+from pymedphys._streamlit.utilities import mosaiq as _mosaiq
+
+# from pymedphys._imports import toml
+
+
+CATEGORY = categories.PLANNING
+TITLE = "Mosaiq to CSV"
+
+LIB_ROOT = pathlib.Path(__file__).parents[3]
+TEST_DATA_DIR = LIB_ROOT.joinpath("tests", "mosaiq", "data")
+
+ALLOWLIST_TABLE_NAMES = [
+    "Ident",
+    "Patient",
+    "TxField",
+    "TxFieldPoint",
+    "Site",
```

Comment:
I would include the SiteSetup table as well.
I'm not sure what the goal is here, but "stuff that goes in a DICOM Plan" includes stuff that comes from SiteSetup

Reply:
Speaking of which... what is the goal here (docstring for the file, docstring for the public methods, and no, it doesn't matter that this is in _experimental, it is really helpful to give us reviewers a hand in understanding what is supposed to happen and why, even if that's the only time anyone ever reads the docstring.

Reply:
> it is really helpful to give us reviewers a hand in understanding what is supposed to happen and why, even if that's the only time anyone ever reads the docstring

Yup, fair call. I'll run through and do that and link to the changes.

Reply:
> I would include the SiteSetup table as well.
> I'm not sure what the goal is here, but "stuff that goes in a DICOM Plan" includes stuff that comes from SiteSetup

In this case I was only actually including tables on an "as-needs" basis. The whole purpose here is to be able to create the following testing suite:

https://github.com/pymedphys/pymedphys/blob/89f4e76e7e579de9e376e59de5f719342e513c9b/lib/pymedphys/tests/mosaiq/test_mimicked_db.py#L39-L163

So tables were only included on an as-needs basis. Given how easy extraction of a new table and its corresponding types is with this `mosaiq_to_csv` tool when there is a PyMedPhys function written that utilises calls to `SiteSetup`, that can be added at that time.